engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/GameManifestProvider.java

```diff
@@ -48,55 +50,91 @@ private GameManifestProvider() {
     /**
      * Generates game manifest with default settings (title, seed) if not specified.
      * Uses default world generator, and modules selection.
-     * @TODO: rewrite/fix it when code will be more stable
      *
-     * @param universeWrapper  contains the universe level properties
-     * @param moduleManager    resolves modules
-     * @param config           provides default module selection, world generator
-     * @return                 game manifest with default settings
+     * @param universeWrapper contains the universe level properties
+     * @param moduleManager   resolves modules
+     * @param config          provides default module selection, world generator
+     * @return game manifest with default settings
+     * @TODO: rewrite/fix it when code will be more stable
      */
-    public static GameManifest createGameManifest(final UniverseWrapper universeWrapper, final ModuleManager moduleManager, final Config config) {
+    public static GameManifest createGameManifest(final UniverseWrapper universeWrapper,
+                                                  final ModuleManager moduleManager,
+                                                  final Config config) {
+
         GameManifest gameManifest = new GameManifest();
-        if (StringUtils.isNotBlank(universeWrapper.getGameName())) {
-            gameManifest.setTitle(universeWrapper.getGameName());
-        } else {
-            gameManifest.setTitle(GameProvider.getNextGameName());
+
+        try {
+            addModulesTo(gameManifest, new DependencyResolver(moduleManager.getRegistry()), config);
+        } catch (DependencyResolutionFailedException e) {
+            logger.error(e.getMessage());
```

Comment:
you can send the exception object itself to the logger without the getMessage, and it will recognize it as an exception and do its exception-formatting thing.

but that can be very verbose, if it's configured to do full stacktraces and whatnot, so disregard this comment if you intentionally want to keep it brief.

Reply:
Thanks for pointing this out :upside_down_face: As I'm aware of the fact that I can simply pass `e` directly I _guess_ I did this on purpose because the output would have been too verbose :thinking: not sure though :see_no_evil: